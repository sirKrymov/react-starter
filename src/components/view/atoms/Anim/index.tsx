// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

interface IProps {
  duration?: string;
  children: ReactNode;
  delay?: string;
  type?: 'mount' | 'show';
}

export function Anim({
  children,
  duration = '0.4s',
  delay = '0.2s',
  type = 'show'
}: IProps): ReactElement {
  return (
    <div
      style={{
        animationDuration: duration,
        animationFillMode: 'both',
        animationDelay: delay,
        animationName: type,
        opacity: 0
      }}
    >
      {children}
    </div>
  );
}
