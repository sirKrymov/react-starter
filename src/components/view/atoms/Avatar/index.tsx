// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import cn from 'classnames';

import './styles.scss';

interface IProps {
  withBorder?: boolean;
  className?: string;
  onClick?(): void;
  avatar?: string;
  size?: 'xs' | 'sm' | 'lg' | number;
  id: string;
}

export function Avatar({
  className,
  onClick,
  withBorder,
  avatar,
  size = 'xs',
  id
}: IProps): ReactElement {
  const avatarClass = cn(
    {
      avatar: true,
      'avatar--with-border': withBorder,
      'avatar--with-click': onClick,
      ...(typeof size === 'string' && { [`avatar--${size}`]: size })
    },
    className
  );

  return (
    <div
      className={avatarClass}
      onClick={onClick ? onClick : undefined}
      style={
        typeof size === 'number'
          ? {
              height: size,
              width: size
            }
          : {}
      }
      id={id}
    >
      <img
        className="img-cover"
        src={
          avatar ||
          'https://f0.pngfuel.com/png/980/886/male-portrait-avatar-png-clip-art.png'
        }
        alt="avatar"
      />
    </div>
  );
}
