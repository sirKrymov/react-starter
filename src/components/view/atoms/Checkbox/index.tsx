// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ChangeEvent, ReactElement } from 'react';

import { IoMdCheckmark } from 'react-icons/io';
import cn from 'classnames';

import './styles.scss';

interface IProps {
  defaultChecked?: boolean;
  onChange(e: ChangeEvent<any>): void;
  className?: string;
  disabled?: boolean;
  value: boolean;
  size?: string;
  name: string;
  id: string;
}

export function Checkbox({
  defaultChecked,
  className,
  onChange,
  disabled,
  value,
  size,
  name,
  id
}: IProps): ReactElement {
  const checkboxClass = cn(
    {
      checkbox: true,
      [`checkbox--size-${size}`]: size,
      'checkbox--disabled': disabled
    },
    className
  );

  return (
    <div className={checkboxClass}>
      <input
        defaultChecked={defaultChecked}
        onChange={onChange}
        disabled={disabled}
        checked={value}
        type="checkbox"
        name={name}
        id={id}
      />

      <label className="checkbox__label" htmlFor={id}>
        <span className="checkbox__icon">
          <IoMdCheckmark />
        </span>
      </label>
    </div>
  );
}
