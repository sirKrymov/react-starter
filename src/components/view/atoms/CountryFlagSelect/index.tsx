// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import cn from 'classnames';

import ReactFlagsSelect from 'react-flags-select';

import './styles.scss';

interface IProps {
  onSelect(country: string): void;
  defaultCountry?: string;
  customConfig?: Record<string, any>;
  countries?: string[];
  className?: string;
  id: string;
}

export function CountryFlagSelect({
  defaultCountry = 'GB',
  customConfig,
  className,
  countries = ['GB'],
  onSelect,
  id
}: IProps): ReactElement {
  const countryFlagSelectClass = cn(
    {
      'country-flag-select': true
    },
    className
  );

  return (
    <div className={countryFlagSelectClass} id={id}>
      <ReactFlagsSelect
        showSelectedLabel={false}
        showOptionLabel={false}
        defaultCountry={defaultCountry}
        countries={countries}
        onSelect={onSelect}
        {...customConfig}
      />
    </div>
  );
}
