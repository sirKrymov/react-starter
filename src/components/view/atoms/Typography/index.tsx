// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

import cn from 'classnames';

import { capitalize } from 'utils';

import './styles.scss';

interface IProps {
  isCapitalize?: boolean;
  component?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'div' | 'p' | 'span';
  paragraph?: boolean;
  uppercase?: boolean;
  className?: string;
  children: ReactNode | string | number;
  variant?:
    | 'subtitle1'
    | 'subtitle2'
    | 'subtitle3'
    | 'inherit'
    | 'caption'
    | 'body1'
    | 'body2'
    | 'body3'
    | 'body4'
    | 'h1'
    | 'h2'
    | 'h3'
    | 'h4'
    | 'h5'
    | 'h6';
  display?: 'initial' | 'block' | 'inline';
  noWrap?: boolean;
  color?:
    | 'textSecondary'
    | 'textPrimary'
    | 'textThirdy'
    | 'textLight'
    | 'secondary'
    | 'textDark'
    | 'primary'
    | 'initial'
    | 'inherit'
    | 'thirdy'
    | 'error'
    | 'valid';
  align?: 'inherit' | 'left' | 'center' | 'right' | 'justify';
  font?: 'roboto';
}

export function Typography({
  isCapitalize,
  component,
  paragraph,
  uppercase,
  className,
  children,
  variant,
  display = 'initial',
  noWrap,
  color = 'textPrimary',
  align = 'left',
  font = 'roboto'
}: IProps): ReactElement {
  const tagMapping = {
    subtitle1: 'h6',
    subtitle2: 'h6',
    subtitle3: 'h6',
    inherit: 'p',
    body1: 'p',
    body2: 'p',
    body3: 'p',
    body4: 'p',
    h1: 'h1',
    h2: 'h2',
    h3: 'h3',
    h4: 'h4',
    h5: 'h5',
    h6: 'h6'
  };
  const Tag =
    component || (paragraph ? 'p' : variant && tagMapping[variant]) || 'span';

  if (isCapitalize && typeof children === 'string')
    children = capitalize(children);

  const typographyClass = cn(
    {
      typography: true,
      [`typography--display-${display}`]: display,
      [`typography--variant-${variant}`]: variant,
      [`typography--color-${color}`]: color,
      [`typography--align-${align}`]: align,
      [`typography--font-${font}`]: font,
      [`typography--uppercase`]: uppercase,
      [`typography--nowrap`]: noWrap
    },
    className
  );

  return <Tag className={typographyClass}>{children}</Tag>;
}
