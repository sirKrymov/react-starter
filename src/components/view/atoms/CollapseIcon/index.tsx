// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import { TiArrowSortedDown } from 'react-icons/ti';
import cn from 'classnames';

import { ReactIcon } from 'components/view/atoms/ReactIcon';

import './styles.scss';

interface IProps {
  onClick?(): void;
  open?: boolean;
  id: string;
}

export function CollapseIcon({ onClick, open, id }: IProps): ReactElement {
  const collapseIconClass = cn({
    'collapse-icon': true,
    'collapse-icon--open': open
  });

  return (
    <div className={collapseIconClass} onClick={onClick} id={id}>
      <ReactIcon size="md" color="on-primary">
        <TiArrowSortedDown />
      </ReactIcon>
    </div>
  );
}
