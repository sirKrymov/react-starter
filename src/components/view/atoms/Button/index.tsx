// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { CSSProperties, ReactElement, ReactNode, memo } from 'react';

import cn from 'classnames';

import { Typography } from 'components/view/atoms/Typography';
import { Loader } from 'components/view/atoms/loaders/Loader';
import { UiSize } from 'components/view/atoms/UiSize';

import './styles.scss';

interface IProps {
  fontWeight?: 'default' | 'bold';
  uppercase?: boolean;
  onClick?(): void | undefined;
  className?: string;
  disabled?: boolean;
  children: ReactNode;
  variant?: 'contained' | 'outlined';
  loading?: boolean;
  height?: 'xs' | 'sm' | 'md' | 'lg' | 'full';
  border?: boolean;
  width?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'full' | 'content';
  color?: 'secondary';
  style?: CSSProperties;
  type?: 'button' | 'submit';
  id: string;
}

function ButtonBase({
  fontWeight = 'bold',
  uppercase,
  className,
  disabled,
  children,
  onClick,
  variant = 'contained',
  loading,
  border,
  height,
  width,
  style,
  color = 'secondary',
  type = 'button',
  id
}: IProps): ReactElement {
  const buttonClass = cn(
    {
      button: true,
      [`button--font-weight-${fontWeight}`]: fontWeight,
      [`button--variant-${variant}`]: variant,
      [`button--padding-${height}`]: height && width === 'content',
      [`button--color-${color}`]: color,
      [`button--disabled`]: disabled,
      [`button--border`]: border,
      'button--loading': loading
    },
    className
  );

  return (
    <UiSize width={width} height={height}>
      <button
        className={buttonClass}
        disabled={disabled}
        onClick={loading || disabled ? undefined : onClick}
        style={style}
        type={type}
        id={id}
      >
        {loading ? (
          <Loader color="on-secondary" />
        ) : (
          <div className="button__content">
            <Typography uppercase={uppercase} variant="inherit" color="inherit">
              {children}
            </Typography>
          </div>
        )}
      </button>
    </UiSize>
  );
}

export const Button = memo(ButtonBase);
