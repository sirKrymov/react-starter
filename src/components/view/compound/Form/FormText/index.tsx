// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

import { Typography } from 'components/view/atoms/Typography';

import './styles.scss';

interface IProps {
  type?: 'simple' | 'title' | 'subtitle';
  text: ReactNode | string;
}

export function FormText({ type, text }: IProps): ReactElement {
  switch (type) {
    case 'title':
      return <Typography variant="h1">{text}</Typography>;

    case 'subtitle':
      return <Typography variant="body1">{text}</Typography>;

    default:
    case 'simple':
      return <Typography variant="body2">{text}</Typography>;
  }
}
