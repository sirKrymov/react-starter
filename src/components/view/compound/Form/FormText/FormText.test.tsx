import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Typography } from '../../../atoms/Typography';
import { FormText } from '../FormText';

describe('<FormText> compound component', () => {
  let shallowComponent: ShallowWrapper;

  const createComponent = (props?: Record<string, any>) => {
    shallowComponent = shallow(<FormText text="Check text" {...props} />);
  };

  describe('WHEN the component loads without prop "type', () => {
    createComponent();

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.find(Typography).prop('variant')).toEqual(
        'body2'
      );
    });
  });

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent({ type: 'title' });
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
