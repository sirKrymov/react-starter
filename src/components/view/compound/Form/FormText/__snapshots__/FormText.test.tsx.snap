// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormText> compound component WHEN the component loads THEN it should match snapshot 1`] = `
<Typography
  variant="h1"
>
  Check text
</Typography>
`;
